cmd/minishift/cmd/root.go

```diff
@@ -251,3 +306,43 @@ func performPostUpdateExecution(markerPath string) error {
 
 	return nil
 }
+
+func ensureAllInstanceConfigPath(ConfigPath string) {
+	configDir := filepath.Dir(ConfigPath)
+	if err := os.MkdirAll(configDir, 0777); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", configDir))
+	}
+}
+
+func createDirForMinishiftRun(minishiftdirs dirs) {
+	dirPaths := reflect.ValueOf(minishiftdirs)
+
+	for i := 0; i < dirPaths.NumField(); i++ {
+		path := dirPaths.Field(i).Interface().(string)
+		if err := os.MkdirAll(path, 0777); err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", path))
+		}
+	}
+}
+
+//If there is no active profiles we need to set
+// minishift as the default profile if minishift profile exists
+func setDefaultActiveProfile() {
```

Comment:
this seems unnecessarily complex. The previous state should have passed something like `""` to `setActiveProfile`

Reply:
@gbraad I differ on this. We need to set the default profile i.e. minishift as an active profile if there is no active profile when `minishift starts`.

Reply:
I still believe this check is confusing, as it is triggered from `root.go`.
The name should be along the lines of `CheckDefaultProfileSthSth()`.

Reply:
@gbraad I am sure about your function name suggestion.  Can you please be more specific?